### ISSUE TITLE
Unbound module Fields

### ISSUE THREAD

**Issue by vbmithr** [janestreet/ppx_csv_conv#1]
Trying to compile

``` ocaml
module Euronext_csv = struct
  type t = {
    name: string;
    isin: string;
    sym: string;
    market: string;
    currency: string;
    o: float;
    h: float;
    l: float;
    c: float;
    ts: string;
    tz: string;
    vol: string;
    cap: string;
  } [@@deriving csv, fields]
end
```

results in:

```
File "src/stsc.ml", line 4, characters 2-258:
Error: Unbound module Fields
Hint: Did you mean Field?
Command exited with code 2.
```

Do I miss anything?

**Comment by ghost** [comment]
The problem is that `csv` depends on `fields`, so `fields` needs to come before `csv`. We have a feature in development to declare dependencies between derivers, but it's not yet ready.

**Comment by vbmithr** [comment]
Any simple way to make this piece of code compile?
When I put ppx_fields_conv before ppx_csv_conv in `_tags`, and use `[@@deriving fields, csv]`, I have the following:

```
File "src/stsc.ml", line 4, characters 2-258:
Error: Unbound module Csvfields
```

**Comment by ghost** [comment]
Ah, it seems I released `ppx_csv_fields` before the library it generates code for... We'll release `csvfields` soon, sorry for that!

**Comment by vbmithr** [comment]
Thank you :)

**Comment by hhugo** [comment]
see resolution of https://github.com/janestreet/ppx_xml_conv/issues/1

**Comment by pveber** [comment]
It seems there is still a bug here (I'm testing with 0.10.0). While
```ocaml
module M = struct
  type t = {
    id : string ;
    count : int ;
  } [@@deriving fields, csv]
end
```
works nicely, 
```ocaml
module M = struct
  type u = {
    id : string ;
    count : int ;
  } [@@deriving fields, csv]
end
```
doesn't work. This is because in both cases `csv` generates code that assumes a `Fields` module exists. While this is true in the former case, this is wrong in the latter, where the module generated by `fields` is called `Fields_of_u`.

**Comment by pveber** [comment]
I stumbled on this bug again on `v0.14.0`, could at least the issue be confirmed and reopened?

**Comment by hhugo** [comment]
I'm reopening it but have not tried to reproduce it.

**Comment by hhugo** [comment]
I can confirm the issue.

**Comment by hhugo** [comment]
We're reviewing a fix

**Comment by aalekseyev** [comment]
This should be fixed in 7ab2b7d.

**Comment by pveber** [comment]
Great, thanks a lot!
